pilot/pkg/networking/plugin/plugin.go

```diff
@@ -109,6 +124,12 @@ type Plugin interface {
 	// added to RDS in the outbound path.
 	OnOutboundRouteConfiguration(in *InputParams, routeConfiguration *xdsapi.RouteConfiguration)
 
+	// OnOutboundRoute is called for every HTTP route in route configurations.
+	OnOutboundRoute(in *InputParams,
+		virtualService *v1alpha3.VirtualService,
+		clusterDescriptors map[string]*ClusterDescriptor,
```

Comment:
fyi: this is an ad-hoc way to decode clusters in the route. The mapping is unstable, so having a structured descriptor is very helpful.